### PR TITLE
Bump golang version to 1.22.2

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: versions.mk [hunk 1]
```diff
@@ -17,8 +17,8 @@ VERSION ?= v0.6.0
 
 vVERSION := v$(VERSION:v%=%)
 
-CUDA_VERSION := 12.3.2
-GOLANG_VERSION := 1.20.5
+CUDA_VERSION := 12.4.1
```
Comment (elezar) [inline versions.mk:20]:
```suggestion
CUDA_VERSION := 12.4.2
```

**Review by elezar — DISMISSED** [review]
Please update to the `12.4.2` version string.

Also, please separate the two changes into two commits.

File: versions.mk (hunk 1 above)
Comment (tariq1890) [reply versions.mk:20]: Typo in my commit message . The latest version is 12.4.1

**Comment by tariq1890** [comment]
@elezar  Sounds good to me!

**Review by elezar — APPROVED** [review]
Looks good now.

I rebased and ran mod tidy.

**Pull request merged**